### PR TITLE
feat: sort account management list of connected chains by user currency balance and chain name

### LINKED ISSUE #6926: Sorting of chains in manage accounts modal
### Overview

Currently the list of chains in the `ManageAccountsModal` are (seemingly) randomly sorted. We should sort them using an intuitive heuristic.

### References and additional details

`src/components/Modals/ManageAccounts/ManageAccountsModal.tsx`
`src/state/slices/common-selectors.ts` -> `selectWalletChainIds`

### Acceptance Criteria

List of connected chains in the `ManageAccountsModal` are sorted using an intuitive heuristic.

Originally raised in:
https://github.com/shapeshift/web/pull/6908#pullrequestreview-2054586834

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

_No response_

### PR DESCRIPTION
**Pull request by woodenfurniture**
## Description

Sorts the list of connected chains in the account management modal by usr currency balance and chain name. 

Replace `selectPortfolioChainIdsSortedUserCurrency` with a corrected version `selectWalletConnectedChainIdsSorted` which sorts the list of connected chainIds based on the sum of all account balances for a given chainId, rather than the most recently seen using a javascript `Set`.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #6926

Moderate risk as this replaces replace `selectPortfolioChainIdsSortedUserCurrency` with a corrected version `selectWalletConnectedChainIdsSorted` which attempts to do that same thing but handling multiple account balances for a given chain correctly.

> What protocols, transaction types or contract interactions might be affected by this PR?

## Screenshots (if applicable)

image

### CODE REVIEW

File: src/state/slices/portfolioSlice/selectors.ts [hunk 1]
```diff
@@ -630,16 +631,6 @@ export const selectCryptoHumanBalanceIncludingStakingByFilter = createCachedSele
...
-export const selectPortfolioChainIdsSortedUserCurrency = createDeepEqualOutputSelector(
-  selectPortfolioAccountsUserCurrencyBalancesIncludingStaking,
-  (userCurrencyAccountBalances): ChainId[] =>
-    Array.from(
-      new Set(
-        Object.keys(userCurrencyAccountBalances).map(accountId => fromAccountId(accountId).chainId),
-      ),
-    ),
-)
-
```
Comment (woodenfurniture) [inline src/state/slices/portfolioSlice/selectors.ts:642]: This appears to be creating an ordered set of chainIds using the `userCurrencyAccountBalances`. The issue here is the the most recent entry will override a previous one, meaning the following issues could arise:
- lots of small balance accounts on a chain with a total higher than 1 large balance on a different chain would be ordered lower
- a large balance followed by a low balance would be ordered lower than a different chain with a moderately sized balance

Comment (woodenfurniture) [reply src/state/slices/portfolioSlice/selectors.ts:642]: The solution proposed in this PR is to sum all of the account balances and sort them based on the total.

File: src/state/slices/portfolioSlice/selectors.ts [hunk 2]
```diff
@@ -1077,3 +1068,43 @@ export const selectEquityTotalBalance = createDeepEqualOutputSelector(
...
+export const selectWalletConnectedChainIdsSorted = createDeepEqualOutputSelector(
+  selectPortfolioTotalBalanceUserCurrencyByChainId,
+  portfolioTotalBalanceUserCurrencyByChainId => {
+    const chainAdapterManager = getChainAdapterManager()
+    return orderBy(
+      Object.entries(portfolioTotalBalanceUserCurrencyByChainId).map(
+        ([chainId, totalBalanceUserCurrency]) => {
+          return {
+            chainId,
+            totalBalanceUserCurrency: Number(totalBalanceUserCurrency),
```
Comment (0xApotheosis) [inline src/state/slices/portfolioSlice/selectors.ts:1101]: It might be safer to `bnOrZero()` here.

**Review by 0xApotheosis — CHANGES_REQUESTED** [review]
Logic looks sane.

Confirmed my EVM chains are now listed in the correct order, vs prod where they are not.

I do, however, notice that my Bitcoin Cash is now incorrectly before my Litecoin - see screenshot below.

Screenshot 2024-05-21 at 9 51 10 AM

File: src/state/slices/portfolioSlice/selectors.ts (hunk 2 above)
Comment (woodenfurniture) [reply src/state/slices/portfolioSlice/selectors.ts:1101]: This guy's only used for sorting of fiat amounts, so precision isn't a concern here. The max value of javascript double should be sufficient to handle any mega fiat balance any user could have so we should be okay?

File: src/state/slices/portfolioSlice/selectors.ts [hunk 3]
```diff
@@ -1096,3 +1087,45 @@ export const selectPortfolioHasWalletId = createSelector(
   (_state: ReduxState, walletId: WalletId) => walletId,
   (storeWalletIds, walletId): boolean => storeWalletIds.includes(walletId),
 )
+
+export const selectPortfolioUserCurrencyTotalBalancesIncludingStakingByChainId =
```
Comment (woodenfurniture) [inline src/state/slices/portfolioSlice/selectors.ts:1091]: new personal best in ridiculously long selector names 💀

**Comment by woodenfurniture** [comment]
> Logic looks sane.
> 
> Confirmed my EVM chains are now listed in the correct order, vs prod where they are not.
> 
> I do, however, notice that my Bitcoin Cash is now incorrectly before my Litecoin - see screenshot below.

Fixed, though I did manage to recreate this issue in develop after a cache clear:
image

**Review by 0xApotheosis — APPROVED** [review]
BCH ordering is great again, the bug above has been resolved 🤝

**Pull request merged**